### ISSUE TITLE
Specify Currency/Chain on ResolveRequest

### ISSUE THREAD

**Issue by sangaman** [ExchangeUnion/xud#831]
When validating the resolve request for swaps we check that the amount matches the expected deal amount, we check cltvdelta, but we don't actually check whether it's the right currency. Theoretically a dishonest peer could attempt to pay us litecoin using the same rHash as for a deal where we expect to receive bitcoin, and xud wouldn't notice the difference because all it sees is the resolve request.

We should add a field to specify the chain or currency to the resolve request and check that this matches the agreed deal when validating a resolve request.

Code reference: https://github.com/ExchangeUnion/xud/blob/master/lib/swaps/Swaps.ts#L557

**Comment by kilrau** [comment]
do you prefer @sangaman to take this over?

**Comment by sangaman** [comment]
Handling it on the xud side is easy and I can certainly do that, but we'll also need a change to raiden to pass the currency/chain value in the request (which also shouldn't be too hard, but I'd need to familiarize myself with the code). lnd I figure we shouldn't bother with since we're going to switch to the hold invoices anyhow.

Having Raiden provide the currency ticker in the request would be great, or the token address would also work if that's easier.

**Comment by offerm** [comment]
This is, more or less,  the request from raiden to XUD:

```

request = {
    'secret_hash': to_hex(secret_request_event.secrethash),
    'amount': secret_request_event.amount,
    'payment_identifier': secret_request_event.payment_identifier,
    'payment_sender': to_hex(secret_request_event.recipient),
    'expiration': secret_request_event.expiration,
    'payment_recipient': to_hex(raiden.address)[2:],
    'reveal_timeout': raiden.config['reveal_timeout'],
    'settle_timeout': raiden.config['settle_timeout'],
}

It includes the payer address and the payee address.
Anything missing?

**Comment by kilrau** [comment]
The token symbol / token address unless it's somehow encoded in `secret_request_event.payment_identifier`

**Comment by offerm** [comment]
Will see if this is available

**Comment by kilrau** [comment]
I moved this up one sprint so that https://github.com/ExchangeUnion/xud/issues/798 can be done first potentially closing this one too.

**Comment by kilrau** [comment]
@offerm is the contract/token address finally available? If not what exactly can you provide to `xud`?

**Comment by offerm** [comment]
Yes, Available now as part of the request.

**Comment by sangaman** [comment]
This can be considered closed once #931 is closed, as it specifies currency requests for raiden, and #923 is merged, as we only create an invoice for the specific currency we are interested in receiving with the new hold invoice approach.

**Comment by kilrau** [comment]
Closing here, since it's already covered by the two issues you mentioned.
